versions.props

```diff
@@ -2,103 +2,82 @@
-    <TelemetryCorrelationVersion>1.0.0</TelemetryCorrelationVersion>
-    <WebInfrastructureVersion>1.0.0</WebInfrastructureVersion>
-    <KubernetesClientVersion>5.0.13</KubernetesClientVersion>
-    <MongoDbClientVersion>2.5.0</MongoDbClientVersion>
+    <!--
+      Non-exposed dependencies, only referenced from test projects or build infrastructure.
+      It's safe to update these at any time, so wildcards are allowed.
+    -->
+
+    <MongoDbClientVersion>2.16.*</MongoDbClientVersion>
+    <MySqlConnectorVersion>2.1.*</MySqlConnectorVersion>
+    <MySqlV8>8.0.*</MySqlV8>
+    <NpgsqlVersion>6.0.*</NpgsqlVersion>
+    <OracleVersion>3.21.*</OracleVersion>
+    <SqlClientVersion>4.8.*</SqlClientVersion>
+    <MicrosoftAzureCosmosVersion>3.28.*</MicrosoftAzureCosmosVersion>
+    <StackExchangeVersion>2.6.*</StackExchangeVersion>
+    <CoverletVersion>3.1.*</CoverletVersion>
+    <FluentAssertionsVersion>6.7.*</FluentAssertionsVersion>
+    <FluentAssertionsJsonVersion>6.1.*</FluentAssertionsJsonVersion>
+    <MockHttpVersion>6.0.*</MockHttpVersion>
+    <MoqVersion>4.18.*</MoqVersion>
+    <TestSdkVersion>17.2.*</TestSdkVersion>
+    <XunitVersion>2.4.*</XunitVersion>
+    <XunitAbstractionsVersion>2.0.*</XunitAbstractionsVersion>
+    <XunitStudioVersion>2.4.*</XunitStudioVersion>
+    <NSubstituteVersion>4.3.*</NSubstituteVersion>
+    <HealthChecksVersion>6.0.*</HealthChecksVersion>
+    <HealthChecksRabbitVersion>3.1.*</HealthChecksRabbitVersion>
+    <SerilogEnrichersThreadVersion>3.1.*</SerilogEnrichersThreadVersion>
+    <SerilogExceptionsVersion>8.2.*</SerilogExceptionsVersion>
+    <SerilogSinksDebugVersion>2.0.*</SerilogSinksDebugVersion>
+    <SonarAnalyzerVersion>8.40.*</SonarAnalyzerVersion>
+    <SourceLinkGitHubVersion>1.1.*</SourceLinkGitHubVersion>
+    <StyleCopVersion>1.1.*</StyleCopVersion>
+    <BenchmarkDotNetVersion>0.13.*</BenchmarkDotNetVersion>
+    <EFCoreTestVersion Condition="'$(TargetFramework)' == 'netcoreapp3.1'">5.0.*</EFCoreTestVersion>
+    <EFCoreTestVersion Condition="'$(TargetFramework)' == 'net6.0'">6.0.*</EFCoreTestVersion>
+    <EFCoreOracleTestVersion Condition="'$(TargetFramework)' == 'netcoreapp3.1'">5.21.*</EFCoreOracleTestVersion>
+    <EFCoreOracleTestVersion Condition="'$(TargetFramework)' == 'net6.0'">6.21.*</EFCoreOracleTestVersion>
 
-    <MySqlCoreVersion>8.0.9-dmr</MySqlCoreVersion>
-    <MySqlConnectorVersion>0.49.3</MySqlConnectorVersion>
-    <MySqlV8>8.0.23</MySqlV8>
-    <NpgsqlVersion>4.0.0</NpgsqlVersion>
-    <SqlClientVersion>4.8.1</SqlClientVersion>
-    <StackExchangeVersion>2.0.593</StackExchangeVersion>
+    <!--
+      Exposed dependencies, observable by Steeltoe consumers.
+      Be cautious with updates:
+      - Wildcards are only allowed when starting work on the next major version, until its release.
+      - Before release of the next major version, pin these down and never change them between minor/patch releases.
```

Comment:
This is a great comment and I like this strategy

Reply:
Glad to hear. I also thought about renaming the first category to `*TestVersion` to make the distinction more explicit. That would make it less likely to accidentally get them mixed up. For example, assume an existing lib in the future needs to reference `HealthChecksTestVersion`, the naming should signal to add `HealthChecksVersion` instead of reuse `HealthChecksTestVersion`. But its still somewhat brittle.

Another approach would be to have test/lib versions in separate files, then reference the lib-versions from project-props and both from test-props.